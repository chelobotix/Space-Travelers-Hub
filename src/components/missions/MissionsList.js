import { v4 as uuidv4 } from 'uuid';
import { useState, useEffect } from "react";
import { useSelector, useDispatch } from "react-redux";
import { increment, fetchMissions } from "../../redux/features/missions/missionsSlice";
import Mission from "./Mission";
import style from './MissionsList.module.css'

const MissionsList = () => {
  const dispatch = useDispatch();
  const state = useSelector((state) => state.missions)


  useEffect(() => {
    dispatch(fetchMissions())

  }, [dispatch])

  if (!state.missionsArray) {
    return <h1>Loading...</h1>;
  }

  return (
    <div>
      <table className={style.table}>
        <thead>
          <tr>
            <th className={style.border}>Mission</th>
            <th className={style.border}>Description</th>
            <th className={style.border}>Status</th>
            <th className={style.border}></th>
          </tr>
        </thead>
        <tbody>
          {state.missionsArray.map((mission) => <tr key={uuidv4()}><Mission mission={mission} /></tr>)}
        </tbody>
      </table>
    </div>
  )
}

export default MissionsList;
