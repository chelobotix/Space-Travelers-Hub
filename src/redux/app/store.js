import { configureStore } from '@reduxjs/toolkit';
import missionsReducer from '../features/missions/missionsSlice'
export const store = configureStore({
  reducer: {
    missions: missionsReducer,
  },
});
