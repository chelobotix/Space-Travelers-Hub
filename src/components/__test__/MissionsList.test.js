import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import MissionsList from '../missions/MissionsList';
import store from '../../redux/app/store';

describe('MissionsList', () => {
  test('MissionsList Tree', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <MissionsList />
      </Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
