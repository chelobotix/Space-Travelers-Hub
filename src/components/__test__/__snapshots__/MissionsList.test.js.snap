// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MissionsList MissionsList Tree 1`] = `
<h1>
  Loading...
</h1>
`;
